### ISSUE TITLE
Regression in 1.2.33: state.tmp.name_node.children is undefined

### ISSUE THREAD

**Issue by dstillman** [Juris-M/citeproc-js#140]
We got a report from a Zotero user using the [Journal of Marketing style](https://www.zotero.org/styles?q=id%3Ajournal-of-marketing), which is [American Marketing Association](https://www.zotero.org/styles?q=id%3Aamerican-marketing-association).

Zotero is currently on citeproc-js 1.3.22, but I traced this regression to 1.2.33, presumably [this commit](https://github.com/Juris-M/citeproc-js/commit/010bd7a9fa37cbb4abbc2ef1ea4475d572de01a6). It still happens on 1.4.6.

Here's the error:

```
zotero(3)(+0000107): TypeError: state.tmp.name_node.children is undefined
    func@chrome://zotero/content/xpcom/citeproc.js:19116:37
    CSL.tokenExec@chrome://zotero/content/xpcom/citeproc.js:2752:21
    CSL.expandMacro/func</<@chrome://zotero/content/xpcom/citeproc.js:2827:28
    CSL.tokenExec@chrome://zotero/content/xpcom/citeproc.js:2752:21
    CSL.expandMacro/func</<@chrome://zotero/content/xpcom/citeproc.js:2827:28
    CSL.tokenExec@chrome://zotero/content/xpcom/citeproc.js:2752:21
    CSL.getCite@chrome://zotero/content/xpcom/citeproc.js:8193:16
    CSL.getBibliographyEntries@chrome://zotero/content/xpcom/citeproc.js:8633:38
    CSL.Engine.prototype.makeBibliography@chrome://zotero/content/xpcom/citeproc.js:8401:11
    getCiteprocBibliography@chrome://zotero/content/xpcom/integration.js:2958:24
    Zotero.Integration.Fields.prototype._updateDocument@chrome://zotero/content/xpcom/integration.js:1179:17
```

In our build, that's this line:

https://github.com/Juris-M/citeproc-js/blob/1f87b7b06d1b2d9a11f20647b6332246b8bb249d/citeproc.js#L19113-L19116

I reproduced this generating a bibliography in a large user document. If the problem here isn't obvious, I can try to narrow it down further.

**Comment by fbennett** [comment]
I was able to build a failing test in the first go. It's not related to the congruency issue flagged in the comment, it's a bit simpler. The bug is triggered with ``partial-each`` substitution whenever names fail and an ordinary field is rendered via ``cs:substitute``. Working on a fix for it now.

**Comment by fbennett** [comment]
This is now fixed in [tag 1.4.7](https://github.com/Juris-M/citeproc-js/tree/1.4.7). The release also contains a fix for [locale-specific title-casing](https://forums.zotero.org/discussion/83567/uppercase-letter-problem-in-turkish) raised earlier on the forums.

**Comment by dstillman** [comment]
Thanks! Now available in the Zotero beta.
